crates/turborepo-ui/src/prefixed.rs

```diff
@@ -94,33 +94,33 @@ enum Command {
 
 /// Wraps a writer with a prefix before the actual message.
 pub struct PrefixedWriter<W> {
-    prefix: StyledObject<String>,
+    prefix: String,
     writer: W,
-    ui: UI,
 }
 
 impl<W> Debug for PrefixedWriter<W> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("PrefixedWriter")
             .field("prefix", &self.prefix)
-            .field("ui", &self.ui)
             .finish()
     }
 }
 
 impl<W: Write> PrefixedWriter<W> {
-    pub fn new(ui: UI, prefix: StyledObject<String>, writer: W) -> Self {
-        Self { ui, prefix, writer }
+    pub fn new(ui: UI, prefix: StyledObject<impl Display>, writer: W) -> Self {
+        let prefix = ui.apply(prefix).to_string();
+        Self { prefix, writer }
     }
 }
 
 impl<W: Write> Write for PrefixedWriter<W> {
     fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
-        let prefix = self.prefix.clone();
-        let prefix = self.ui.apply(prefix);
-        let prefix_bytes_written = self.writer.write(prefix.to_string().as_bytes())?;
+        self.writer.write_all(self.prefix.as_bytes())?;
```

Comment:
Switched to `write_all` since we weren't checking if all bytes got written to the underlying writer.